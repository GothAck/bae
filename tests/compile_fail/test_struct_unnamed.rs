use bae::{test_utils::parse_attrs, FromAttributes};
use quote::quote;
use syn::{LitInt, LitStr};

#[derive(FromAttributes, Debug)]
struct MyAttr(
    LitStr,
    Option<LitInt>,
    Option<LitInt>,
    Option<()>,
    Option<()>,
);

fn main() {
    let tokens = quote! { #[my_attr(str = "123", optional_int_given = 456, switch_given)] };

    let attrs = parse_attrs(tokens).unwrap();
    let attr = MyAttr::from_attributes(&attrs).unwrap();

    assert_eq!(attr.0.value(), "123");
    assert_eq!(attr.1.unwrap().to_string(), "456");
    assert_eq!(attr.2, None);
    assert!(attr.3.is_some());
    assert!(attr.4.is_none());
}
