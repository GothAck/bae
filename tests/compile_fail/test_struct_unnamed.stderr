error: Field without a name
 --> tests/compile_fail/test_struct_unnamed.rs:7:5
  |
7 |     LitStr,
  |     ^^^^^^

error[E0599]: no function or associated item named `from_attributes` found for struct `MyAttr` in the current scope
  --> tests/compile_fail/test_struct_unnamed.rs:18:24
   |
6  | / struct MyAttr(
7  | |     LitStr,
8  | |     Option<LitInt>,
9  | |     Option<LitInt>,
10 | |     Option<()>,
11 | |     Option<()>,
12 | | );
   | |__- function or associated item `from_attributes` not found for this
...
18 |       let attr = MyAttr::from_attributes(&attrs).unwrap();
   |                          ^^^^^^^^^^^^^^^ function or associated item not found in `MyAttr`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `from_attributes`, perhaps you need to implement it:
           candidate #1: `FromAttributes`
